pkg/mcp/loader.go

```diff
@@ -508,6 +509,89 @@ func (sm *SessionManager) updatedMCPPodName(ctx context.Context, url, id string,
 	}
 }
 
+// LaunchTemporaryInstance creates a temporary MCP server from a catalog entry, lists its tools,
+// then shuts it down and returns the tool preview data.
+func (sm *SessionManager) LaunchTemporaryInstance(ctx context.Context, catalogEntryManifest otypes.MCPServerCatalogEntryManifest, userURL string, configEnv map[string]string) ([]otypes.MCPServerTool, error) {
```

Comment:
This logic is very similar to the logic in `toolsForServer` in `pkg/api/handlers/mcp.go`, especially the part where we parse the input parameters. Can we find a way of not duplicating this admittedly complex logic?

Also, the name of this method is a bit misleading.

Reply:
Yeah I'll see if I can dry it up